### PR TITLE
feat(sdk-coin-sui): added sui explain transaction

### PR DESCRIPTION
**Pull request by linyah9912**
Ticket: BG-59020

Implemented Sui explain transaction and added missing verifyTransaction and fromRawTransaction unit tests

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/sdk-coin-sui/test/unit/sui.ts [hunk 1]
```diff
@@ -56,13 +56,64 @@ describe('SUI:', function () {
...
+    });
+  });
+
+  describe('Explain Transaction: ', () => {
+    it('should explain a transfer transaction', async function () {
+      const explainedTransaction = await basecoin.explainTransaction({
+        txHex: testData.TRANSFER_TX,
+      });
+      explainedTransaction.should.deepEqual({
+        displayOrder: ['id', 'outputs', 'outputAmount', 'changeOutputs', 'changeAmount', 'fee', 'type'],
```
Comment (margueriteblair) [inline modules/sdk-coin-sui/test/unit/sui.ts:86]: what does outputAmount and changeOutputs refer to here? Does SUI have these properties in Object model?

Comment (margueriteblair) [reply modules/sdk-coin-sui/test/unit/sui.ts:86]: If SUI doesn't have concept of change, I'd say remove those properties, and make SUI's own native tx explainTransaction() as opposed to using baseTransaction.explainTransaction()

Comment (linyah9912) [reply modules/sdk-coin-sui/test/unit/sui.ts:86]: SUI doesn't have these properties, but these fields are not optional under `interface ITransactionExplanation`

Comment (linyah9912) [reply modules/sdk-coin-sui/test/unit/sui.ts:86]: Making SUI's own SuiTransactionExplanation also works and makes sense, but I checked all other coins and they return these properties even if they don't have them like SUI. So I'm wondering if I should follow the current behaviour or just make it Sui specific.

Comment (margueriteblair) [reply modules/sdk-coin-sui/test/unit/sui.ts:86]: Interesting...weird that all other coins would have these too, I'd say it's probably fine then. Something to maybe revisit as a backlog item for altcoins team later on :)

**Review by margueriteblair — APPROVED** [review]

File: modules/sdk-coin-sui/test/unit/sui.ts (hunk 1 above)
Comment (linyah9912) [reply modules/sdk-coin-sui/test/unit/sui.ts:86]: agree, it'd be something to revisit and clean up later!

**Pull request merged**